pkg/crc/api/handlers.go

```diff
@@ -0,0 +1,125 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/code-ready/crc/cmd/crc/cmd/config"
+	crcConfig "github.com/code-ready/crc/pkg/crc/config"
+	"github.com/code-ready/crc/pkg/crc/constants"
+	"github.com/code-ready/crc/pkg/crc/logging"
+	"github.com/code-ready/crc/pkg/crc/machine"
+	"github.com/code-ready/crc/pkg/crc/validation"
+	"github.com/code-ready/crc/pkg/crc/version"
+)
+
+func statusHandler(_ ArgsType) string {
+	statusConfig := machine.ClusterStatusConfig{Name: constants.DefaultName}
+	clusterStatus, err := machine.Status(statusConfig)
+	if err != nil {
+		logging.Error(err.Error())
+	}
+	s, err := json.Marshal(clusterStatus)
+	if err != nil {
+		return "Failed"
+	}
+	return string(s)
+}
+
+func stopHandler(_ ArgsType) string {
+	stopConfig := machine.StopConfig{
+		Name:  constants.DefaultName,
+		Debug: true,
+	}
+	commandResult, err := machine.Stop(stopConfig)
+	if err != nil {
+		logging.Error(err.Error())
+	}
+	s, err := json.Marshal(commandResult)
+	if err != nil {
+		return "Failed"
+	}
+	return string(s)
+}
+
+func startHandler(_ ArgsType) string {
+	startConfig := machine.StartConfig{
+		Name:          constants.DefaultName,
+		BundlePath:    crcConfig.GetString(config.Bundle.Name),
+		VMDriver:      crcConfig.GetString(config.VMDriver.Name),
+		Memory:        crcConfig.GetInt(config.Memory.Name),
+		CPUs:          crcConfig.GetInt(config.CPUs.Name),
+		NameServer:    crcConfig.GetString(config.NameServer.Name),
+		GetPullSecret: getPullSecretFileContent,
+		Debug:         true,
+	}
+	fmt.Println(crcConfig.GetString(config.PullSecretFile.Name))
+	status, err := machine.Start(startConfig)
```

Comment:
from here it is practically the same pattern as the previous command `status`, so I expect this to be repeated. can you abstract this?

Reply:
Okay, i think the common statements can be taken out and made into function, will modify this

Reply:
you could use a generic function to pass a `func` in that performs the action, while the generic code would perform error check and logging.

Reply:
So i extracted the common steps of converting the struct to a json string as a separate function.
Please take a look again if you want this to be further abstracted, which would be difficult as each function takes different types of arguments.

Reply:
it can be condensed more, but looking better. acceptable